tests/ptp/tbc.go

```diff
@@ -0,0 +1,152 @@
+// Copyright 2023 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ptp
+
+import (
+	"time"
+	"os"
+	"os/exec"
+	"bufio"
+	"regexp"
+	"strings"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/redhat-partner-solutions/vse-sync-testsuite/pkg/images"
+	//"github.com/redhat-partner-solutions/vse-sync-testsuite/pkg/pods"
+
+	"github.com/redhat-partner-solutions/vse-sync-testsuite/pkg/clients"
+	"github.com/redhat-partner-solutions/vse-sync-testsuite/pkg/collectors/devices"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	PTP_GROUP_LENGTH = 4
+)
+const (
+	PTP_METRICS = 4
+)
+
+const (
+	ptpLogsProcessed = "/tmp/data.csv"
+)
+
+const (
+	ptpLogs = "/tmp/data.log"
+)
+
+func Readln(r *bufio.Reader) (string, error) {
+  var (isPrefix bool = true
+       err error = nil
+       line, ln []byte
+      )
+  for isPrefix && err == nil {
+      line, isPrefix, err = r.ReadLine()
+      ln = append(ln, line...)
+  }
+  return string(ln),err
+}
+
+func decodePtp4line(tstamp string, sync_metrics string, processed_file *os.File ) error {
+	//log.Info(sync_metrics)
+	re := regexp.MustCompile(`-?\d+`)
+	ptp4l_sample_data := re.FindAllString(sync_metrics,-1)
+	//log.Info("ALL DATA before join ", ptp4l_sample_data)
+	if len(ptp4l_sample_data) == PTP_METRICS {
+		//write timestamp, offset with master, state, frequency, and path delay
+		ptp4l_all_sample_data := []string{tstamp,ptp4l_sample_data[0],ptp4l_sample_data[1],ptp4l_sample_data[2],ptp4l_sample_data[3]}
+		join := strings.Join(ptp4l_all_sample_data,",")
+		_, err := processed_file.WriteString(join + "\n")
+		if (err != nil){
+			return err
+		}
+	}
+	return nil
+}
+
+func parsePtpLogs(filename string, filename_dst string) error {
```

Comment:
Move these helpers to a new file, or after the tests in this one, to make it easier to see what the tests are doing.

Reply:
Fully Agree. Also note that these "helpers" will be reused in other test suites.